File: investai/investai/backend/models/usuario.py
from .base import ModeloBase, db


class Usuario(ModeloBase):
    __tablename__ = "usuario"

    nome = db.Column(db.String(120), nullable=False)
    email = db.Column(db.String(120), nullable=False, unique=True)
    perfil_risco = db.Column(db.String(20), nullable=False, default="conservador")
    renda_mensal = db.Column(db.Float, nullable=False, default=0.0)

    movimentacoes = db.relationship(
        "Movimentacao", back_populates="usuario", cascade="all, delete-orphan"
    )
    investimentos = db.relationship(
        "Investimento", back_populates="usuario", cascade="all, delete-orphan"
    )
    metas = db.relationship(
        "Meta", back_populates="usuario", cascade="all, delete-orphan"
    )

    def to_dict(self):
        return {
            "id": self.id,
            "nome": self.nome,
            "email": self.email,
            "perfil_risco": self.perfil_risco,
            "renda_mensal": self.renda_mensal,
        }

    def __repr__(self):
        return f"<Usuario {self.nome}>"

File: investai/frontend_flutter/lib/models/usuario.dart
class Usuario {
  final int? id;
  final String nome;
  final String email;
  final String perfilRisco;
  final double rendaMensal;

  Usuario({
    this.id,
    required this.nome,
    required this.email,
    required this.perfilRisco,
    required this.rendaMensal,
  });

  factory Usuario.fromJson(Map<String, dynamic> json) {
    return Usuario(
      id: json['id'],
      nome: json['nome'],
      email: json['email'],
      perfilRisco: json['perfil_risco'] ?? 'conservador',
      rendaMensal: (json['renda_mensal'] ?? 0).toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'nome': nome,
      'email': email,
      'perfil_risco': perfilRisco,
      'renda_mensal': rendaMensal,
    };
  }
}
